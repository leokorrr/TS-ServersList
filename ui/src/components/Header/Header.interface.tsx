export default interface HeaderInterface {
    numberOfServers: number;
    dataToSearch: Function;
}