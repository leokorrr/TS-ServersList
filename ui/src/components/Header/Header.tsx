import React, {useState, useEffect} from 'react'
import SearchInput from '../SearchInput/SearchInput'
import HeaderInterface from './Header.interface'

const Header: React.FC<HeaderInterface> = ({numberOfServers, dataToSearch}) => {
    const [valueFromInput, setValueFromInput] = useState('')
    const onSearch = (value: string) => {
        setValueFromInput(value)
    }
    useEffect(() => {
        dataToSearch(valueFromInput)
    }, [valueFromInput, dataToSearch])
    return (
        <header>
            <div className="header">
                <div className="header__col">
                    <h2 className="header__title">Servers</h2>
                    <h3 className="header__subtitle">Number of elements: {numberOfServers}</h3>
                </div>
                <div className="header__col">
                    <SearchInput onSearch={onSearch}/>
                </div>
            </div>
        </header>
    )
}

export default Header