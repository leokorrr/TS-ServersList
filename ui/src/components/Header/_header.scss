.header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 1.5rem;
    margin-top: 3.9rem;

    &__title {
        font-weight: 600;
        font-size: 2.1rem;
        margin: 0 0 1.1rem 0;
    }

    &__subtitle {
        font-weight: 400;
        font-size: 1.5rem;
        margin: 0;
    }

    &__col {
        &:last-of-type {
            display: flex;
            flex-direction: column-reverse;
        }
    }
}