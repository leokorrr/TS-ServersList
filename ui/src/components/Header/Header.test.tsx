import React from 'react'
import Header from './Header'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

const numberOfServers = 16
const dataToSearch = () => {}
configure({ adapter: new Adapter() })
it("Header rendered without crashing", () => {
    shallow(<Header numberOfServers={numberOfServers} dataToSearch={dataToSearch}/>);
})
it("should render initial layout", () => {
    const component = shallow(<Header numberOfServers={numberOfServers} dataToSearch={dataToSearch}/>);
    expect(component.getElements()).toMatchSnapshot();
})