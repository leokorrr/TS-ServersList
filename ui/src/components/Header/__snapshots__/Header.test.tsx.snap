// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
Array [
  <header>
    <div
      className="header"
    >
      <div
        className="header__col"
      >
        <h2
          className="header__title"
        >
          Servers
        </h2>
        <h3
          className="header__subtitle"
        >
          Number of elements: 
          16
        </h3>
      </div>
      <div
        className="header__col"
      >
        <SearchInput
          onSearch={[Function]}
        />
      </div>
    </div>
  </header>,
]
`;
